lib/yara/scanner.rb

```diff
@@ -0,0 +1,81 @@
+module Yara
+  class Scanner
+    class NotCompiledError < StandardError; end
+
+    ERROR_CALLBACK = proc do |error_level, file_name, line_number, rule, message, user_data|
+      # noop
+    end
+
+    SCAN_FINISHED = 3
+
+    # Public: Initializes instance of scanner. Under the hood this calls yr_initialize, then
+    # creates a pointer, then calls yr_compiler_create with that pointer.
```

Comment:
I think we can update this comment

```suggestion
    # Public: Initializes instance of scanner. Under the hood this creates a pointer,
    # then calls yr_compiler_create with that pointer.
```